plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,648 @@
+ 
+
+##### Mitigations for Honest Exit Bond Slashing
+
+###### Bond Sharing
+
+One way to partially mitigate this attack is for each user who piggybacks to cover some portion of `exit bond`.
+This cuts the per-person value of `exit bond` proportionally to the number of users who have piggybacked.
+
+
+###### Small Exit Bond
+
+The result of the above attack is that users may not exit from an in-flight transaction if the gas cost of exiting plus the value of `exit bond` is greater than the value of their input or output.
+We can reduce the impact of this attack by minimizing the gas cost of exiting and the value of `exit bond`.
+Gas cost should be highly optimized in any case, so the value of `exit bond` is of more importance.
+
+`exit bond` is necessary to incentivize challenges.
+However, we believe that challenges can be sufficiently incentivized if `exit bond` simply covers the gas cost of challenging.
+Observations from the Bitcoin and Ethereum ecosystems suggest that sufficiently many nodes will verify transactions without a direct in-protocol incentive to do so.
+Our system requires only a single node be properly incentivized to challenge, and it’s likely that many node operators will have strong external incentives.
+Modeling the “correct” size of the exit bond is an ongoing area of research.
+
+
+###### Timeouts
+
+We can add timeouts to each transaction (“must be included in the chain by block X”) to reduce number of vulnerable transactions at any point in time.
+This will probably also be necessary from a user experience point of view, as we don’t want users to accidentally sign a double-spend simply because the first transaction hasn’t been processed yet.
+
+
+## Alice-Bob Scenarios
+
+### Alice & Bob are honest and cooperating:
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator begins withholding blocks while `TX1` is still in-flight.
+Neither Alice nor Bob know if the transaction has been included in a block.
+4. Someone with access to `TX1` (Alice, Bob, or otherwise) starts an exit referencing `TX1` and places `exit bond`.
+5. Alice and Bob piggyback onto the exit and each place `piggyback bond`.
+6. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+7. After period 2, Bob receives the value of `UTXO2`.
+All bonds are refunded.
+
+
+### Mallory tries to exit a spent output:
+
+1. Alice spends `UTXO1` in `TX1` to Mallory, creating `UTXO2`.
+2. `TX1` is included in block `N`.
+3. Mallory spends `UTXO2` in `TX2`.
+4. Mallory starts an exit referencing `TX1` and places `exit bond`.
+5. Mallory piggybacks onto the exit and places `piggyback bond`.
+6. In period 2, someone reveals `TX2` spending `UTXO2`.
+This challenger receives Mallory’s `piggyback bond`.
+7. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+8. After period 2, Mallory’s `exit bond` is refunded.
```

Comment:
note explicitly that no-one exits any of the UTXOs perhaps? If yes, then that applies to all scenarios

Reply:
Added to each scenario.